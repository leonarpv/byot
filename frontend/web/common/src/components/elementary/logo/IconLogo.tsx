import * as React from 'react';
import {makeStyles, Theme} from '@material-ui/core';
import {Logo} from './Logo';
import {WithStyles} from '../../../types/WithStyles';
import {Rectangle} from '@byot-frontend/common/src/data-manipulation/Rectangle';
import {Constants} from '@byot-frontend/common/src/types/Constants';

interface Props extends WithStyles<typeof styles> {
  height?: number;
}

const styles = (theme: Theme) => ({
  root: (props: Props) => ({
    overflow: 'hidden',
  }),
});
const useStyles = makeStyles(styles);

export const IconLogo: React.FC<Props> = ({height = 100, ...props}: Props) => {
  const styles = useStyles(props);
  // 192x320
  const originalSizeRectangle = new Rectangle(192, 320);
  const wrappingSize = Rectangle.calculateScaled({
    originWidth: originalSizeRectangle.width,
    originHeight: originalSizeRectangle.height,
    targetHeight: height,
  });
  return (
    <div
      className={styles.root}
      style={{
        width: `${wrappingSize.width}px`,
        height: `${wrappingSize.height}px`,
      }}>
      <Logo height={(height * Constants.LOGO_SIZE.height) / originalSizeRectangle.height} />
    </div>
  );
};
