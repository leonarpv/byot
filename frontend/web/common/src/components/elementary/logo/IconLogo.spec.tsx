import * as React from 'react';
import {render} from '@testing-library/react';
import {IconLogo} from './IconLogo';

describe('<IconLogo/>', () => {
  it('should render', () => {
    const {container} = render(<IconLogo />);
    expect(container).toMatchSnapshot();
  });
});
