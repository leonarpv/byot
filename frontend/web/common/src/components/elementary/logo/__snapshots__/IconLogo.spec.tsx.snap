// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconLogo/> should render 1`] = `
<div>
  <div
    class="makeStyles-root-1 makeStyles-root-2"
    style="width: 60px; height: 100px;"
  >
    <div
      class="MuiBox-root MuiBox-root-6 makeStyles-root-3 makeStyles-root-5"
    >
      <img
        class="makeStyles-img-4"
        src="logo.svg"
      />
    </div>
  </div>
</div>
`;
